src/main.rs

```diff
@@ -222,25 +222,25 @@ async fn handle_request(
 ) -> Result<Response<Body>, RequestError> {
     trace!("Incoming request: {:?}", request);
 
-    let request_path = request.uri().path().to_string();
-
-    let (method, m) = match request.method() {
-        &HttpMethod::DELETE => (Method::Delete, "DELETE"),
-        &HttpMethod::GET => (Method::Get, "GET"),
-        &HttpMethod::PATCH => (Method::Patch, "PATCH"),
-        &HttpMethod::POST => (Method::Post, "POST"),
-        &HttpMethod::PUT => (Method::Put, "PUT"),
+    let (method, m) = match *request.method() {
+        HttpMethod::DELETE => (Method::Delete, "DELETE"),
+        HttpMethod::GET => (Method::Get, "GET"),
+        HttpMethod::PATCH => (Method::Patch, "PATCH"),
+        HttpMethod::POST => (Method::Post, "POST"),
+        HttpMethod::PUT => (Method::Put, "PUT"),
         _ => {
             error!("Unsupported HTTP method in request, {}", request.method());
             return Err(RequestError::InvalidMethod {
-                method: request.method().to_owned(),
+                method: request.method().clone(),
```

Comment:
Can avoid one allocation here
```suggestion
                method: request.into_parts().0.method,
```